scripts/components/server-side-render/server-side-render.js

```diff
@@ -0,0 +1,170 @@
+		}
+	}
+
+	fetch(props) {
+		if (!this.isStillMounted) {
+			return;
+		}
+		if (null !== this.state.response) {
+			//this.setState({ response: null, prevResponse: this.state.response });
+			this.setState(state => (
+				{
+					response: null,
+					prevResponse: state.response
+				}
+			));
+		}
+		const { block, attributes = null, urlQueryArgs = {} } = props;
+
+		const path = rendererPath(block, attributes, urlQueryArgs);
+		// Store the latest fetch request so that when we process it, we can
+		// check if it is the current request, to avoid race conditions on slow networks.
+		const fetchRequest = (this.currentFetchRequest = apiFetch({ path })
+			.then((response) => {
+				if (
+					this.isStillMounted &&
+					fetchRequest === this.currentFetchRequest &&
+					response
+				) {
+					this.setState({ response: response.rendered });
+				}
+			})
+			.catch((error) => {
+				if (
+					this.isStillMounted &&
+					fetchRequest === this.currentFetchRequest
+				) {
+					this.setState({
+						response: {
+							error: true,
+							errorMsg: error.message,
+						},
+					});
+				}
+			}));
+		return;
+	}
+
+	render() {
+
+		const { right, top, unit } = this.props.spinnerLocation;
+		const response = this.state.response;
+		const prevResponse = this.state.prevResponse;
+		let prevResponseHTML = "";
+		if (prevResponse !== null) {
+			prevResponseHTML = `<div style="position:relative;"><div style="z-index:10000;position:absolute;right:${right}${unit};top:${top}${unit}"><span class="components-spinner"></span></div>${prevResponse}</div>`;
+		}
+
+		const {
+			className,
+			EmptyResponsePlaceholder,
+			ErrorResponsePlaceholder,
+		} = this.props;
+
+		if (response === '') {
+			return (
+				<EmptyResponsePlaceholder
+					response={response}
+					{...this.props}
+				/>
+			);
+		} else if (!response) {
+			return (
+				<Fragment>
+					<RawHTML key="html" className={className}>
+						{prevResponseHTML}
+					</RawHTML>
+				</Fragment>
+			);
+		} else if (response.error) {
+			return (
+				<ErrorResponsePlaceholder
+					response={response}
+					{...this.props}
+				/>
+			);
+		}
+
+		return (
+			<RawHTML key="html" className={className}>
+				{response}
+			</RawHTML>
+		);
+	}
+}
+
+ServerSideRender.defaultProps = {
+	spinnerLocation: { right: 0, top: 10, unit: 'px' },
+	EmptyResponsePlaceholder: ({ className }) => (
+		<Placeholder className={className}>
+			{__('Block rendered as empty.')}
+		</Placeholder>
+	),
+	ErrorResponsePlaceholder: ({ response, className }) => {
+		const errorMessage = sprintf(
+			// translators: %s: error message describing the problem
+			__('Error loading block: %s'),
+			response.errorMsg
+		);
+		return (
+			<Placeholder className={className}>{errorMessage}</Placeholder>
+		);
+	},
+	LoadingResponsePlaceholder: ({ className }) => {
+		return (
+			<Placeholder className={className}>
+				<Spinner />
+			</Placeholder>
+		);
+	},
+};
+
+export default ServerSideRender;
```

Comment:
It says no newline at the end. Can you configure your vscode to automatically add newlines at the end of the file on saving?

Why: https://unix.stackexchange.com/a/18789
How: https://stackoverflow.com/a/44704969/629127

Reply:
Didn't know that was a setting, thanks!